snmp-observ-lib/signals/memory.libsonnet

```diff
@@ -0,0 +1,142 @@
+      huawei: 'hwEntityMemUsage',
+      juniper: 'jnxOperatingBuffer',
+      mikrotik: self.generic,
+      netgear: 'agentSwitchCpuProcessMemAvailable',
+      qtech: 'switchMemoryBusy',
+      tplink: 'tpSysMonitorMemoryUtilization',
+      ubiquiti_airos: 'memTotal',
+    },
+    signals: {
+      memoryUsage: {
+        name: 'Memory utilization',
+        description: 'Memory utilization.',
+        type: 'gauge',
+        unit: 'percent',
+        sources: {
+          generic: {
+            expr: |||
+              hrStorageUsed{%(memorySelector)s, %%(queriesSelector)s}/hrStorageSize{%(memorySelector)s, %%(queriesSelector)s}*100
+            ||| % { memorySelector: this.memorySelector },
+          },
+          arista_sw: self.generic,
+          brocade_fc: {
+            expr: 'swMemUsage{%(queriesSelector)s}',
+          },
+          brocade_foundry: {
+            expr: 'snAgGblDynMemUtil{%(queriesSelector)s}',
+          },
+          cisco: {
+            expr: |||
+              # cisco
+              # ciscoMemoryPoolType="1" - Processor
+              ciscoMemoryPoolUsed{ciscoMemoryPoolType="1", %(queriesSelector)s}
+              /
+              (ciscoMemoryPoolUsed{ciscoMemoryPoolType="1", %(queriesSelector)s} + ciscoMemoryPoolFree{ciscoMemoryPoolType="1", %(queriesSelector)s}) * 100
+            |||,
+            // aggKeepLabels: ['ciscoMemoryPoolName'],
+          },
+          dell_network: {
+            expr: 'dellNetCpuUtilMemUsage{%(queriesSelector)s}',
+          },
+          dlink_des: {
+            expr: 'agentDRAMutilization{%(queriesSelector)s}',
+          },
+          extreme: {
+            expr: |||
+              (extremeMemoryMonitorSystemTotal{%(queriesSelector)s}-extremeMemoryMonitorSystemFree{%(queriesSelector)s})/extremeMemoryMonitorSystemTotal{%(queriesSelector)s}*100
+            |||,
+          },
+          eltex_mes: {
+            expr: 'issSwitchCurrentRAMUsage{%(queriesSelector)s}',
+          },
+          // https://my.f5.com/manage/s/article/K44014003
+          f5_bigip: {
+            expr: |||
+              # f5_bigip
+              (sysGlobalTmmStatMemoryUsed{%(queriesSelector)s}+sysGlobalHostOtherMemoryUsed{%(queriesSelector)s})
+              /
+              (sysGlobalTmmStatMemoryTotal{%(queriesSelector)s} + sysGlobalHostOtherMemoryTotal{%(queriesSelector)s}) * 100
+            |||,
+          },
+          fortigate: {
+            expr: self.generic,
+          },
+          hpe: {
+            expr: |||
+              # hpe
+              hpLocalMemAllocBytes{%(queriesSelector)s}
+              /
+              hpLocalMemTotalBytes{%(queriesSelector)s} * 100
+            |||,
+          },
+          huawei: {
+            expr: 'hwEntityMemUsage{%(queriesSelector)s}',
+          },
+          juniper: {
+            expr: 'jnxOperatingBuffer{jnxOperatingContentsIndex="9", %(queriesSelector)s}',
+            // aggKeepLabels: ['jnxOperatingDescr'],
```

Comment:
Minor nit, if we don't want these aggregation labels they shouldn't be commented without a reason given

Reply:
added comments. I want to use it in next iteration.